packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -317,24 +317,6 @@ async function test() {
   return promise;
 }
     `,
-
-    // optional chaining
-    `
-async function test() {
-  declare const returnsPromise: () => Promise<void> | null;
-  await returnsPromise?.();
-  returnsPromise()?.then(
-    () => {},
-    () => {},
-  );
-  returnsPromise()
-    ?.then(() => {})
-    ?.catch(() => {});
-  returnsPromise()?.catch(() => {});
-  returnsPromise()?.finally(() => {});
```

Comment:
aren't all of these valid still?
They all either are `await`ed, or have a `catch`/`finally` to close off the promise.